coderd/database/dbauthz/dbauthz.go

```diff
@@ -4871,6 +4924,20 @@ func (q *querier) UpdateUserRoles(ctx context.Context, arg database.UpdateUserRo
 	return q.db.UpdateUserRoles(ctx, arg)
 }
 
+func (q *querier) UpdateUserSecret(ctx context.Context, arg database.UpdateUserSecretParams) (database.UserSecret, error) {
+	// First get the secret to check ownership
+	secret, err := q.db.GetUserSecret(ctx, arg.ID)
```

Comment:
I'm still new to the rbac stuff, is this pattern common, where we go a query to get an object so we can check the permissions before doing the query we want to do?

Reply:
I wonder if we could avoid this query by having the user id as part of the secret id or something.

Reply:
@bcpeinhardt it is a common pattern because we do not have enough context to do the authz check before the call. 

This is unfortunate, and there are sometimes where the context is fully present beforhand. One day we might want to optimize these cases :shrug: